lib/Frontend/Frontend.cpp

```diff
@@ -383,6 +383,11 @@ void CompilerInstance::setUpDiagnosticOptions() {
   }
   Diagnostics.setDiagnosticDocumentationPath(
       Invocation.getDiagnosticOptions().DiagnosticDocumentationPath);
+  if (!Invocation.getDiagnosticOptions().LocalizationCode.empty()) {
```

Comment:
Since I'm checking now to see if `LocalizationCode.empty()`. `DiagnosticEngine:: setLocalization` won't be called till I'm sure that I have a LocalizationCode and a LocalizationPath. Is the assertion now necessary, and if it's where will I `assert` and on what exactly? 😅
cc @LucianoPAlmeida

Reply:
To me since we already validate this early, I think we don't need to do this here... we can simply just assert inside `setLocalization`.

Reply:
So I will call `assert` in setLocalization to see if the path and the locale code not empty, right?

Reply:
Yes, that is correct. To clarify `assert` is useful to validate assumptions of the `setLocalization` itself, although you know that it's currently called in a particular way doesn't mean that it wouldn't change in the future and possibly by somebody who didn't realize that both locale and path shouldn't be empty before calling it. In situations like that `assert` would provide a clear signal about what is expected...